spec/lucky/cookies/cookie_jar_spec.cr

```diff
@@ -1,93 +1,132 @@
 require "../../spec_helper"
 
 describe Lucky::CookieJar do
-  it "gets and sets with indifferent access" do
-    jar = Lucky::CookieJar.new
+  it "sets and gets with indifferent access" do
+    jar = Lucky::CookieJar.empty_jar
 
     jar.set(:symbol_key, "symbol key")
     jar.set("string_key", "string key")
 
-    jar.get(:symbol_key).value.should eq("symbol key")
-    jar.get("symbol_key").value.should eq("symbol key")
-    jar.get("string_key").value.should eq("string key")
-    jar.get(:string_key).value.should eq("string key")
+    jar.get(:symbol_key).should eq("symbol key")
+    jar.get("symbol_key").should eq("symbol key")
+    jar.get("string_key").should eq("string key")
+    jar.get(:string_key).should eq("string key")
   end
 
-  describe "#set" do
-    it "sets a cookie's expiration date by default" do
-      jar = Lucky::CookieJar.new
-
-      jar.set(:message, "help i'm trapped in a cookie jar")
-
-      expiration = jar.get(:message).expires.not_nil!
-      # dirty hack because I can't get a time mocking lib to work
-      # this works when I test just this file but not in the full suite
-      # my guess is because the time set in the cookie jar is a constant that
-      # is set at compile time, which ends taking more than 1 second to compile
-      # 1 minute difference for a year seems reasonable for now
-      expiration.should be_close(1.year.from_now, 1.minute)
+  it "sets and gets raw HTTP::Cookie object with indifferent access" do
+    value = "Nestle Tollhouse"
+    jar = Lucky::CookieJar.empty_jar
+
+    jar.set_raw(:cookie, value)
+    jar.set_raw(:symbol, "symbol value")
+
+    jar.get_raw(:cookie).should be_a(HTTP::Cookie)
+    jar.get_raw("symbol").value.should eq("symbol value")
+    jar.get_raw(:cookie).value.should eq(value)
+    jar.get_raw("cookie").value.should eq(value)
+    jar.get_raw?(:cookie).not_nil!.value.should eq(value)
+    jar.get_raw?("cookie").not_nil!.value.should eq(value)
+    jar.get_raw?(:missing).should be_nil
+    jar.get_raw?("missing").should be_nil
+  end
+
+  it "catches values with old or incorrect keys and returns nil" do
+    jar_with_old_secret = Lucky::CookieJar.empty_jar
+    Lucky::Server.temp_config(secret_key_base: "a" * 32) do
+      jar_with_old_secret.set(:name, "value")
     end
+    value_encrypted_with_old_jar = jar_with_old_secret.get_raw(:name).value
+    jar = Lucky::CookieJar.empty_jar
+
+    jar.set_raw(:name, value_encrypted_with_old_jar)
```

Comment:
Good thought. I was going to do that but I was reading about it and it seems most major frameworks reset the cookie if it has the wrong key. That way if your key leaks you can change it. Then the next time a user comes they’ll be signed out instead of getting an error. 

It’s also not a super helpful error because if the key is wrong there’s nothing you can do about it from the app side of things. By setting to nil it’ll just work in the next request.